pkg/sql/exec_log.go

```diff
@@ -347,6 +362,23 @@ func (p *planner) maybeLogStatementInternal(
 	if shouldLogToAdminAuditLog {
 		p.logEventsOnlyExternally(ctx, eventLogEntry{event: &eventpb.AdminQuery{CommonSQLExecDetails: execDetails}})
 	}
+
+	if telemetryLoggingEnabled {
+		smoothQPS := telemetryMetrics.expSmoothQPS()
+		shouldSampleEventToTelemetry := p.stmt.AST.StatementType() == tree.TypeDML && smoothQPS > qpsThreshold
+		// If we DO NOT need to sample the event, log immediately to the telemetry
+		// channel. Otherwise, log the event to the telemetry channel if it has been
+		// sampled.
+		// TODO(esr): add the effective sample rate to the SampledQuery event. Reference: https://github.com/cockroachdb/cockroach/issues/69653
```

Comment:
Now that you're coming back, you should add yourself, not esr =)

Reply:
Will do :)